src/test/java/org/terasology/web/io/JsonSessionTest.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.web.io;
+
+import com.google.common.base.Preconditions;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonNull;
+import org.junit.Test;
+import org.terasology.context.Context;
+import org.terasology.context.internal.ContextImpl;
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.identity.PublicIdentityCertificate;
+import org.terasology.identity.storageServiceClient.BigIntegerBase64Serializer;
+import org.terasology.network.ClientComponent;
+import org.terasology.network.ClientInfoComponent;
+import org.terasology.registry.CoreRegistry;
+import org.terasology.web.authentication.AuthenticationFailedException;
+import org.terasology.web.authentication.AuthenticationHandshakeHandler;
+import org.terasology.web.authentication.ClientAuthenticationMessage;
+import org.terasology.web.authentication.HandshakeHello;
+import org.terasology.web.client.HeadlessClient;
+import org.terasology.web.client.HeadlessClientFactory;
+
+
+import java.math.BigInteger;
+import java.util.Arrays;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class JsonSessionTest {
+
+    private static class AuthenticationHandshakeHandlerMock implements AuthenticationHandshakeHandler {
+        private boolean nextResult;
+
+        @Override
+        public HandshakeHello initServerHello() {
+            return null;
+        }
+
+        @Override
+        public byte[] authenticate(ClientAuthenticationMessage authenticationMessage) throws AuthenticationFailedException {
+            Preconditions.checkNotNull(authenticationMessage.getClientHello());
+            Preconditions.checkNotNull(authenticationMessage.getSignature());
+            if (!nextResult) {
+                throw new AuthenticationFailedException(true);
+            }
+            return new byte[]{0, 0, 0};
+        }
+    }
+
+    private static final Gson GSON = new GsonBuilder()
```

Comment:
Maybe another reason to create a shared instance of GSON for the HTTP/WS marshalling.